### PR TITLE
Try to add support for more modern arrow versions

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Adapt https://github.com/snowflakedb/snowflake-connector-python/pull/418 for conda-forge

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
@xhochy Okay got this to build fully with additional builds for arrow 2.0 and 3.0 (skipping 1.0).  Additionally this builds the cython arrow extensions for windows as well.

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Review by xhochy — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,18 +38,22 @@ requirements:
...
+    - arrow-cpp
+    - pyarrow
+    - numpy
   run:
     - python
-    - pyarrow 0.17.1
+    {% if pyarrow == '0.17.1' %}
+    - pyarrow {{ pyarrow }}
+    {% else %}
+    - {{ pin_compatible('pyarrow', max_pin='x') }}
```
Comment (mariusvniekerk) [inline recipe/meta.yaml:55]: this seems to put a pin for 3.0 in both cases??

Comment (mariusvniekerk) [reply recipe/meta.yaml:55]: nm seems like a weird conda index bug locally

Comment (xhochy) [reply recipe/meta.yaml:55]: Rebuilt this locally and the pins looks fine, merging this.

**Pull request merged**